### ISSUE TITLE
subnetMatch throws exception on empty range

### ISSUE THREAD

**Issue by anton-kotenko** [whitequark/ipaddr.js#14]
Hi, I use ipaddr.js to check if ip address belongs to some of the dynamically built ip address range. This range can be empty, and in this case subnetMatch method throws exception. 
Example to reproduce the problem:

````
var ipaddr = require('ipaddr.js'),
    ip = ipaddr.parse('1.2.3.4');

try {
    ipaddr.subnetMatch(ip, {subnet: []}, false);
} catch (e) {
    console.log('Why am i failing???', e, e.stack);
}
```` 
Of course i can solve problem with just one if statement, but such behaviour seems strange to me (i expect it just to return false). 
The source of such behaviour is next lines https://github.com/whitequark/ipaddr.js/blob/master/src/ipaddr.coffee#L34-L35
So question: is this a bug, or it was done intentionally?

**Comment by whitequark** [comment]
That's a bug.

**Comment by whitequark** [comment]
Released in 1.0.2.
